### DISCUSSION TITLE
Allow strings as values for Tabs active prop

### DISCUSSION THREAD

**Discussion by mcmxcdev** [mantinedev/mantine#1661] [Feature requests]
I need to have buttons that look like this: https://mantine.dev/core/tabs/#add-your-styles-with-styles-api which act as replacement for standard radio buttons in our case, since the designers don't like the look of default radio buttons and I couldn't find a `UnstyledRadio` component.

Chakra-UI has a `useRadio` hook, see: https://chakra-ui.com/docs/components/radio/usage#custom-radio-buttons

I am using https://mantine.dev/core/tabs/#controlled-tabs but discovered that the `active` prop only accepts numeric values and `onChange` gives back the tab index rather than the value that is passed into.

It would be great to be able to work with string values, e.g. `male/female/other` rather than having them represented as `0/1/2`. This is how I would imagine it to simulate radio button behavior:

```
<InputWrapper label="Sex" size="xl">
  <TabButtonGroup
    {...rest}
    active={value}
    onTabChange={onChange}
  >
    <Tabs.Tab value="male" label="Male" />
    <Tabs.Tab value="female" label="Female" />
  </TabButtonGroup>
</InputWrapper>
```

**Comment by rtivital** [comment]
This is already implemented in https://v5.mantine.dev/changelog/5-0-0/

**Reply by mcmxcdev** [reply]
Amazing, v5 works like a charm, thank you.

Although its in alpha state, I didn't encounter any bugs yet! :smile:

**Comment by mcmxcdev** [comment]
@rtivital Unfortunately, I have further requests for this issue:

Although we can now use `Tabs` component with string values just fine, it doesn't make sense for e.g. simple `"yes"/"no"` cases. 
We would like to use booleans there.

I can also imagine that in other cases, numbers would indeed be useful as well.

Can you consider changing `export declare type TabsValue = string | null;` to be `export declare type TabsValue = string | number | boolean | null;` ?
